### PR TITLE
[REG-2184] Initial docs for Bot Actions and End Criteria

### PR DESCRIPTION
**Pull request by vontell**
Adds initial documentation for botAction and endCriteria. Includes data formats and examples.

Note to reviewers: In terms of priorities, I wanted to make sure that this information is at least in these docs, with the understanding that it will improve over time. I don't want us to spend so much time making these perfect in case things change, but at least have enough of a reference for developers using the platform.

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

File: docs/core-concepts/bot-sequences/actions.mdx [hunk 1]
```diff
@@ -1,9 +1,624 @@
...
+                "position": {"x": ..., "y": ...}, // A coordinate indicating the position to click, where X and Y are integers
+                "worldPosition": null, // An optional coordinate indicating the position to click in world space, which is often null.
+                "leftButton": ..., // A boolean indicating whether the left mouse button should be pressed
+                "middleButton": ..., // A boolean indicating whether the middle mouse button should be pressed
+                "rightButton": ..., // A boolean indicating whether the right mouse button should be pressed
+                "forwardButton": ..., // A boolean indicating whether the forward mouse button should be pressed
+                "backButton": ..., // A boolean indicating whether the back mouse button should be pressed
+                "scroll": {"x": ..., "y": ...}, // An optional scroll amount, where X and Y are integers
+                "clickedObjectNormalizedPaths": [...], // An optional list of normalized paths to the objects that were clicked
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/actions.mdx:110]: We should add to the comment that these are used to adjust click coordinates when there are variances in resolution scaling or object position from run to run

File: docs/core-concepts/bot-sequences/actions.mdx [hunk 2]
```diff
@@ -1,9 +1,624 @@
...
+
+```json
+// The `botAction.data` format
+{
+    "startTime": 0.0, // The time offset at which the action should be performed from the start of the segment
+    "inputData": {
+        "mouse": [ // Any number of mouse inputs can be provided
+            {
+                "startTime": ..., // A number indicating the time offset at which the action should be performed from the start of the segment (in seconds)
+                "screenSize": {"x": ..., "y": ...}, // A width and height indicating the size of the screen, which is used to scale the position of the mouse when the screen size is different from the screen size at the time the inputs were recorded
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/actions.mdx:101]: should say 'linearly' scale so it is clear what kind of scaling we do

File: docs/core-concepts/bot-sequences/actions.mdx [hunk 3]
```diff
@@ -1,9 +1,624 @@
...
+    "endCriteria": [...],
+    "botAction": {
+        "type": "InputPlayback",
+        "data": {
+            "startTime": 0.0,
+            "inputData": {
+                "keyboard": [],
+                "mouse": [
+                    {"startTime":0.0,"screenSize":{"x":1920,"y":1080},"position":{"x":942,"y":500},"worldPosition":null,"leftButton":false,"middleButton":false,"rightButton":false,"forwardButton":false,"backButton":false,"scroll":{"x":0,"y":0},"clickedObjectNormalizedPaths":["UI Canvas/Start Button"]},
+                    {"startTime":0.1,"screenSize":{"x":1920,"y":1080},"position":{"x":942,"y":500},"worldPosition":null,"leftButton":true,"middleButton":false,"rightButton":false,"forwardButton":false,"backButton":false,"scroll":{"x":0,"y":0},"clickedObjectNormalizedPaths":["UI Canvas/Start Button"]}
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/actions.mdx:177]: we should include a third entry here with the leftButton:false to show the click release

File: docs/core-concepts/bot-sequences/actions.mdx [hunk 4]
```diff
@@ -1,9 +1,624 @@
...
+    "description": "Moves the mouse to a position and clicks it",
+    "endCriteria": [...],
+    "botAction": {
+        "type": "InputPlayback",
+        "data": {
+            "startTime": 0.0,
+            "inputData": {
+                "keyboard": [],
+                "mouse": [
+                    {"startTime":0.0,"screenSize":{"x":1920,"y":1080},"position":{"x":942,"y":500},"worldPosition":null,"leftButton":false,"middleButton":false,"rightButton":false,"forwardButton":false,"backButton":false,"scroll":{"x":0,"y":0},"clickedObjectNormalizedPaths":["UI Canvas/Start Button"]},
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/actions.mdx:176]: it is good that this is here, we should add a comment stating that it is best practice to have a mouse position input before the click input .  Without this, Unity's input handling can cause some un-expected behaviour with the click especially when the prior mouse position was over a different input component in the UI.

File: docs/core-concepts/bot-sequences/actions.mdx [hunk 5]
```diff
@@ -1,9 +1,624 @@
...
+particular data and information from the code of your game, such as tapping into an existing navigation system.
+
+**Data Format**
+
+`type` is set to `Behaviour`.
+
+```json
+// The `botAction.data` format
+{
+    "behaviourFullName": "...", // The full class path of the MonoBehaviour to run
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/actions.mdx:197]: fully qualified

File: docs/core-concepts/bot-sequences/actions.mdx [hunk 6]
```diff
@@ -1,9 +1,624 @@
...
+    "maxRuntimeSeconds": ... // The maximum amount of time to run the behaviour for in seconds before it is stopped
+}
+```
+
+The Behaviour itself needs to be defined in the codebase of your game where it can be found during runtime. It should be
+a class that inherits from `MonoBehaviour`. In order to terminate the behaviour, you can use `Destroy(this)` to destroy the
+behaviour instance.
+
+**Example: Running a pathfinding behaviour for at max 30 seconds**
+
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/actions.mdx:207]: We need to add a blurb about IRGBotSegmentActionBehaviour to make Behaviours pausible.


We also need to add a short blurb explaining that the Update loop of your behaviour will run as normal until you destroy it or the timer limit is reached at which point the SDK will destroy it

File: docs/core-concepts/bot-sequences/actions.mdx [hunk 7]
```diff
@@ -1,9 +1,624 @@
...
+The `MonkeyBot` action type runs a chaos monkey bot for a given amount of time. This can be useful for testing the robustness of your game.
+This Monkey Bot will randomly select actions to perform from a list of possible actions, and will perform them at a given interval. These
+action are found via a code analysis of the game, which must be run before you use this segment type.
+
+**Setup**
+
+In order to setup the monkey bot, you need to run a code analysis of the game. This can be done from the **Regression Games > Configure Bot Actions** menu.
+
+Once you click that menu option, a panel will appear. Click the "Analyze Actions" button, which will perform an analysis of your game and
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/actions.mdx:271]: We should add a sentence about them filtering out actions.

File: docs/core-concepts/bot-sequences/actions.mdx [hunk 8]
```diff
@@ -1,9 +1,624 @@
...
+    "name": "Run monkey bot",
+    "description": "Runs a monkey bot to click on random objects while ignoring the quit actions",
+    "endCriteria": [...],
+    "botAction": {
+        "type": "MonkeyBot",
+        "data": {
+            "actionInterval": 0.5,
+            "actionSettings": {
+                "DisabledActionPaths": [
+                    "Unity UI/Button/Unity.Multiplayer.Samples.BossRoom.Visual.UISettingsCanvas.OnClickQuitButton",
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/actions.mdx:303]: We aren't giving them any context for how to get these strings, we need to add / explain that

File: docs/core-concepts/bot-sequences/actions.mdx [hunk 9]
```diff
@@ -1,9 +1,624 @@
...
+    - A resource:// path to a .bytes TextAsset in one of your project's Resources folders that is a JPG or PNG file saved with a .bytes file extension.  This can be used if you do not want Unity to import your image as a Texture.
+- `withinRect` - An optional (can be null/undefined) field to limit the search area to a specific pixel region of the current frame.  The SDK will linearly tranform the supplied `rect` to fit the current resolution using the `screenSize` as the initial reference resolution.
+    - `screenSize` - The reference resolution in pixels which defines the screen space that `rect` is defined within.
+    - `rect` - The position (x=0, y=0 is bottom left) and size (width, height) of the rectangle that must contain the supplied image data.  The values are defined in pixels.
+- `actions` - The list of mouse actions to take at the center point of the returned `rect`.
+
+**Notes**
+
+The `CVImage` type is still in an experimental phase and may provide inconsistent or unexpected results in many situations.
+- Multiple matches of the specified image within a frame may provide incorrect or inconsistent result bounds.
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/actions.mdx:334]: We need to add a blurb about how images are compared in grey scale

File: docs/core-concepts/bot-sequences/actions.mdx [hunk 10]
```diff
@@ -1,9 +1,624 @@
...
+}
+```
+A bot action has a `type` and a `data` field. The `type` indicates the type of action to be taken, and the `data` field contains
+the parameters for that action. For example, a bot action to click on a coordinate may look like this:
+
+```json
+{
+    "name": "Click the play button",
+    "description": "Clicks the play button in the main menu",
+    "keyFrameCriteria": [...],
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/actions.mdx:36]: should say 'endCriteria'

File: docs/core-concepts/bot-sequences/actions.mdx [hunk 11]
```diff
@@ -1,9 +1,624 @@
...
+## Structure of a Bot Action
+
+Bot Actions are defined in Bot Segments within the `botAction` field. A segment may only contain one bot action.
+
+```json
+// Segment structure
+{
+    "name": "Click the play button",
+    "description": "Clicks the play button in the main menu",
+    "keyFrameCriteria": [...],
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/actions.mdx:20]: should say 'endCriteria'

File: docs/core-concepts/bot-sequences/actions.mdx [hunk 12]
```diff
@@ -1,9 +1,624 @@
...
+/9j/4AAQ...<rest of the base64 encoded image data>...VLj3ieS/9k=
+```
+
+**Example: Using base64 byte[]**
+```json
+{
+    "name": "CV Image Action: Click Menu Change Profile Button",
+    "description": "Moves the mouse over the change profile button, then clicks and releases on the button. Criteria waits for the action to complete.",
+    "endCriteria": [
+        {"type":"ActionComplete", "transient":false, "data":{}}
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/actions.mdx:358]: This is the first time `endCriteria` has been filled in.  Either exclude this everywhere (Eesh) or fill this in everywhere and provide a quick explainer at the start.

File: docs/core-concepts/bot-sequences/actions.mdx [hunk 13]
```diff
@@ -1,9 +1,624 @@
...
+- `textMatchingRule` - One of `Matches` or `Contains`
+    - `Matches` - Each word in the provided `text` must be found as a whole word in the results.  This is a very exact matching rule and can sometimes lead to inconsistent results in frames with poor in game lighting on the text or low text contrast.
+    - `Contains` - Each word in the provided `text` must be found as a part of a word in the results.  This is a looser matching rule and if often used instead of `Matches` for more stable results.  For example "Time" would match to "Timer", "Time", "Times", "Timed", "Time", etc in the frame.  This may not seem ideal, but in most game situations, the contrast or text layout isn't always clearly identifiable and extra or incorrect letters may consistently be found for the text you are looking for in a specific frame of the game.  When you encounter those situations, you can adjust your game to give better text clarity/contrast to your users and/or you can utilize `Contains` instead of `Matches`.
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/actions.mdx:410]: "Time" is in the list 2x

File: docs/core-concepts/bot-sequences/actions.mdx [hunk 14]
```diff
@@ -1,9 +1,624 @@
...
+```
+
+### RestartGame
+
+This action type restarts the game. This should only be used as the 'last' segment in your sequence or segment list, and is useful for getting your game
+ready for another test.
+
+There is an interface you can implement to override the default behaviour of restarting the game. If this interface is not implemented when 
+the `RestartGame` Bot Action is used in a build of your game **outside the UnityEditor**, then the default action will be to call 
+`SceneManager.LoadScene(sceneBuildIndex: 0, mode: LoadSceneMode.Single)`.  **In the UnityEditor**, the action will **always** be to stop and re-enter play mode;
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/actions.mdx:566]: We also destroy all 'dontdestroyonload' game objects before loading scene 0

File: docs/core-concepts/bot-sequences/end-criteria.mdx [hunk 15]
```diff
@@ -4,6 +4,440 @@ sidebar_label: 'End Criteria'
...
+## Structure of End Criteria
+
+End Criteria are defined in Bot Segments within the `endCriteria` field. A segment may contain multiple end criteria.
+
+```json
+// Segment structure
+{
+    "name": "Wait for the player to appear",
+    "description": "Waits for the player to appear on the screen",
+    "keyFrameCriteria": [
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/end-criteria.mdx:19]: replace all `keyFrameCriteria` with `endCriteria`

File: docs/core-concepts/bot-sequences/end-criteria.mdx [hunk 16]
```diff
@@ -4,6 +4,440 @@ sidebar_label: 'End Criteria'
...
+:::
+
+## End Criteria Types
+
+There are many different types of end criteria that can be used in a bot segment, each useful in their own way.
+The list of supported types are as follows:
+
+* `NormalizedPath`: This end criteria waits for a specific game object to exist or not exist in the scene.
+* `PartialNormalizedPath`: Similar to `NormalizedPath`, but supports partial matches of the path string.
+* `UIPixelHash`: This end criteria waits and passes if the pixels on the screen change.
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/end-criteria.mdx:68]: This is currently only supported/used to detect changes in 3rd party UI systems like GameFace

File: docs/core-concepts/bot-sequences/end-criteria.mdx [hunk 17]
```diff
@@ -4,6 +4,440 @@ sidebar_label: 'End Criteria'
...
+* `CVText`: This end criteria waits for a specific text to appear on the screen using computer vision.
+* `CVImage`: This end criteria waits for a specific image to appear on the screen using computer vision.
+* `CVObjectDetection`: This end criteria waits for a specific object to appear on the screen using computer vision.
+* `ActionComplete`: This end criteria waits for the provided bot actions to complete.
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/end-criteria.mdx:72]: action (singular)

File: docs/core-concepts/bot-sequences/end-criteria.mdx [hunk 18]
```diff
@@ -4,6 +4,440 @@ sidebar_label: 'End Criteria'
...
+    "botAction": {...}
+}
+```
+
+:::info
+There are additional fields that can be set on a bot action and many of these `inputData` fields, such as an apiVersion. You can see the full
+specification in the [source code](https://github.com/Regression-Games/RGUnityBots/blob/main/src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/KeyFrameCriteria.cs).
+:::
+
+## End Criteria Types
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/end-criteria.mdx:61]: We need to discuss that by default, all criteria are AND'd together.

We do also have support for explicit `And` and `Or` criteria.

File: docs/core-concepts/bot-sequences/end-criteria.mdx [hunk 19]
```diff
@@ -4,6 +4,440 @@ sidebar_label: 'End Criteria'
...
+            }
+        }
+    ],
+    "botAction": {...}
+}
+```
+
+### UIPixelHash
+
+The `UIPixelHash` end criteria waits and passes if the pixels on the screen change. This automatically passes if this is the first frame of the test.
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/end-criteria.mdx:169]: Should add a line about how this is only used for 3rd party UI systems like GameFace

File: docs/core-concepts/bot-sequences/end-criteria.mdx [hunk 20]
```diff
@@ -4,6 +4,440 @@ sidebar_label: 'End Criteria'
...
+            }
+        }
+    ],
+    "botAction":{}
+}
+```
+
+### ActionComplete
+
+The `ActionComplete` end criteria is used to indicate that the segment is completed once all actions of this segment have been completed.
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/end-criteria.mdx:422]: action (Singular).. a bot segment only defines a single botAction

File: sidebars.js [hunk 21]
```diff
@@ -46,8 +46,8 @@ const sidebars = {
           label: "Bot Sequences",
           items: [
             "core-concepts/bot-sequences/getting-started-with-bot-sequences-and-segments",
-            // "core-concepts/bot-sequences/actions",
-            // "core-concepts/bot-sequences/end-criteria"
+            "core-concepts/bot-sequences/actions",
+            "core-concepts/bot-sequences/end-criteria"
```
Comment (RG-nAmKcAz) [inline sidebars.js:50]: Let's put endCriteria before actions (see notes above about how actions refer to endCriteria as though the user already understands them)

File: docs/core-concepts/bot-sequences/end-criteria.mdx [hunk 22]
```diff
@@ -4,6 +4,440 @@ sidebar_label: 'End Criteria'
...
+specification in the [source code](https://github.com/Regression-Games/RGUnityBots/blob/main/src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/KeyFrameCriteria.cs).
+:::
+
+## End Criteria Types
+
+There are many different types of end criteria that can be used in a bot segment, each useful in their own way.
+The list of supported types are as follows:
+
+* `NormalizedPath`: This end criteria waits for a specific game object to exist or not exist in the scene.
+* `PartialNormalizedPath`: Similar to `NormalizedPath`, but supports partial matches of the path string.
```
Comment (RG-nAmKcAz) [inline docs/core-concepts/bot-sequences/end-criteria.mdx:67]: We should note that `NormalizedPath` is far more performant than `PartialNormalizedPath` and should be preferred when possible.

**Review by RG-nAmKcAz — CHANGES_REQUESTED** [review]
Great writeup overall

needs small corrections or content additions for accuracy/clarity

File: docs/core-concepts/bot-sequences/actions.mdx (hunk 7 above)
Comment (vontell) [reply docs/core-concepts/bot-sequences/actions.mdx:271]: Added a note about this. The experience here is actually pretty bad.... since Monkey Bots are configured in the botAction JSON now, these settings in the panel are completely ignored (and this is a low hanging fruit I wanted to address soon).

File: docs/core-concepts/bot-sequences/actions.mdx (hunk 8 above)
Comment (vontell) [reply docs/core-concepts/bot-sequences/actions.mdx:303]: Now covered in the note added above, and also added a comment to the JSON.

File: docs/core-concepts/bot-sequences/actions.mdx (hunk 9 above)
Comment (vontell) [reply docs/core-concepts/bot-sequences/actions.mdx:334]: Added

File: docs/core-concepts/bot-sequences/actions.mdx (hunk 12 above)
Comment (vontell) [reply docs/core-concepts/bot-sequences/actions.mdx:358]: Good catch, forgot to remove during the copy paste. Fixed!

File: docs/core-concepts/bot-sequences/actions.mdx (hunk 13 above)
Comment (vontell) [reply docs/core-concepts/bot-sequences/actions.mdx:410]: Fixed

File: docs/core-concepts/bot-sequences/actions.mdx (hunk 14 above)
Comment (vontell) [reply docs/core-concepts/bot-sequences/actions.mdx:566]: Added

File: docs/core-concepts/bot-sequences/end-criteria.mdx (hunk 15 above)
Comment (vontell) [reply docs/core-concepts/bot-sequences/end-criteria.mdx:19]: Fixed

File: docs/core-concepts/bot-sequences/end-criteria.mdx (hunk 18 above)
Comment (vontell) [reply docs/core-concepts/bot-sequences/end-criteria.mdx:61]: Do you have an example segment somewhere that uses Or and And explicitly? I actually skipped those primarily because I wasn't entirely sure how they were setup in a sample. I'm looking at the data format in the code now and see it's not too complicated so I just added those to this doc now. Let me know if they look incorrect!

File: docs/core-concepts/bot-sequences/end-criteria.mdx (hunk 22 above)
Comment (vontell) [reply docs/core-concepts/bot-sequences/end-criteria.mdx:67]: Added a note

File: docs/core-concepts/bot-sequences/end-criteria.mdx (hunk 16 above)
Comment (vontell) [reply docs/core-concepts/bot-sequences/end-criteria.mdx:68]: Added note.

File: docs/core-concepts/bot-sequences/end-criteria.mdx (hunk 17 above)
Comment (vontell) [reply docs/core-concepts/bot-sequences/end-criteria.mdx:72]: Fixed

File: docs/core-concepts/bot-sequences/end-criteria.mdx (hunk 19 above)
Comment (vontell) [reply docs/core-concepts/bot-sequences/end-criteria.mdx:169]: Added note

File: docs/core-concepts/bot-sequences/end-criteria.mdx (hunk 20 above)
Comment (vontell) [reply docs/core-concepts/bot-sequences/end-criteria.mdx:422]: Fixed!

File: sidebars.js (hunk 21 above)
Comment (vontell) [reply sidebars.js:50]: Adjusted 👍

**Review by vontell — COMMENTED** [review]
Thanks for all the comments! I believe all of them have been addressed now.

**Review by RG-nAmKcAz — APPROVED** [review]

File: docs/core-concepts/bot-sequences/actions.mdx [hunk 23]
```diff
@@ -1,9 +1,659 @@
...
+        ]
+    }
+}
+```
+
+**Notes**
+
+- You can use the `startTime` and `endTime` fields to control the timing of actions. For instance, you can hold shift and press a key to result in a capital letter.
+- If two actions have the same `startTime`, the action that appears earlier in the list will be performed first.
+- You can omit an `endTime` of a keyboard action to have the action continue until it is disabled using using an action that only sets the `endTime`.
```
Comment (abeizer) [inline docs/core-concepts/bot-sequences/actions.mdx:128]:
```suggestion
- You can omit an `endTime` of a keyboard action to have that action continue until it is disabled by a future action which only sets the `endTime`.
```

Honestly still confusing to read, even after the suggestion. I don't want to spend too much time making it read well though unless we think it would be highly useful.

File: docs/core-concepts/bot-sequences/actions.mdx [hunk 24]
```diff
@@ -1,9 +1,659 @@
...
+        }
+    }
+}
+```
+
+### MonkeyBot
+
+The `MonkeyBot` action type runs a chaos monkey bot for a given amount of time. This can be useful for testing the robustness of your game.
+This Monkey Bot will randomly select actions to perform from a list of possible actions, and will perform them at a given interval. These
+action are found via a code analysis of the game, which must be run before you use this segment type.
```
Comment (abeizer) [inline docs/core-concepts/bot-sequences/actions.mdx:306]:
```suggestion
actions are found via a code analysis of the game, which must be run before you use this segment type.
```

**Review by abeizer — APPROVED** [review]
I can think of some really good ways to organize all of this information when we're at the point of needing nicer-looking docs. For now I think this will be really helpful and answer 90% of PW's questions

File: docs/core-concepts/bot-sequences/actions.mdx [hunk 25]
```diff
@@ -1,9 +1,659 @@
...
+There are additional fields that can be set on a bot action and many of these `inputData` fields, such as an apiVersion. You can see the full
+specification in the [source code](https://github.com/Regression-Games/RGUnityBots/blob/cd68ed0f4b10567eb652477717338d92b4f4571e/src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/BotAction.cs).
+:::
+
+## Action Types
+
+There are many different types of actions that can be used in a bot segment, each useful in their own way.
+The list of supported types are as follows:
+
+* `InputPlayback`: This action plays back a sequence of mouse and keyboard inputs.
```
Comment (addisonbgross) [inline docs/core-concepts/bot-sequences/actions.mdx:75]:
```suggestion
* `InputPlayback`: This action plays back a set of mouse and keyboard inputs.
```

**Review by addisonbgross — APPROVED** [review]

File: docs/core-concepts/bot-sequences/actions.mdx (hunk 25 above)
Comment (vontell) [reply docs/core-concepts/bot-sequences/actions.mdx:75]: Updated, changed to "series" though to at least give the sense that ordering does matter (in the case where the start times are the same)
